realsense/plugin.cpp

```diff
@@ -9,20 +9,42 @@
 
 using namespace ILLIXR;
 
+#define T26X
```

Comment:
Is there a preferred method for getting configuration information from config yaml to the plugin code? What I have found so far seems to use env vars set by the runner script, such as 
config.yaml
`enable_alignment: False`
main.py
`enable_offload_flag = config["enable_offload"]`
`enable_offload{ILLIXR::str_to_bool(ILLIXR::getenv_or("ILLIXR_OFFLOAD_ENABLE", "False"))}` and
`if (enable_offload) ...`
plugin.cpp

Reply:
Currently, the best way is to follow the `offload_data` plugin example as you've found. This requires a few steps:
- [ ] Add your variable to `runner/config_schema.yaml` (like [`enable_offload`](https://github.com/ILLIXR/ILLIXR/blob/508fd83e9361e8b011c598787892ffe1e982592b/runner/config_schema.yaml#L128)).
- [ ] In each action ([`load_native`](https://github.com/ILLIXR/ILLIXR/blob/508fd83e9361e8b011c598787892ffe1e982592b/runner/runner/main.py#L84), [`load_tests`](https://github.com/ILLIXR/ILLIXR/blob/508fd83e9361e8b011c598787892ffe1e982592b/runner/runner/main.py#L140), [`load_monado`](https://github.com/ILLIXR/ILLIXR/blob/508fd83e9361e8b011c598787892ffe1e982592b/runner/runner/main.py#L167)):
    - [ ] Load the value from the config (like [`enable_offload`](https://github.com/ILLIXR/ILLIXR/blob/508fd83e9361e8b011c598787892ffe1e982592b/runner/runner/main.py#L144))
    - [ ] Optionally process the value
    - [ ] Set the value in the environment (like [`ILLIXR_ENABLE_OFFLOAD`](https://github.com/ILLIXR/ILLIXR/blob/508fd83e9361e8b011c598787892ffe1e982592b/runner/runner/main.py#L158))
    - [ ] Read, process, and set your value within your client code (like [`ILLIXR_ENABLE_OFFLOAD`](https://github.com/ILLIXR/ILLIXR/blob/508fd83e9361e8b011c598787892ffe1e982592b/offload_data/plugin.cpp#L33))

For your use case, it looks like you would like to do a few things:
1. Set the camera type,
1. Set some parameters based on the camera type, and
1. Do some logic based on the camera type.

Since are passing in parameters to the program using environment variables, we have to be careful with statically determined variables and compilation:
1. The camera type will probably have to be string environment variable that we can set in the relevant files in `configs`. This value will have to be loaded in the constructor block of the plugin because the environment variables are set _after_ compilation.
2. Either in the constructor, or in the body in the context where each variable is used, the camera property should be read to determine the values.
3. In the body where logic is muxed based on the camera property, the logic needs to be handled dynamically (without static compilation guards).

This is a tedious process we are hoping to streamline with the #198 patch, but we have some other patches on the critical path first.

Reply:
@e3m3 Thanks for the detailed reply, I'll probably get this updated later today.

Reply:
I will not have time to review this properly before next week, but why do we need to set configs/parameters/defines at all for this? Can we not automatically detect which camera has been plugged in using [query_devices](https://github.com/IntelRealSense/librealsense/blob/master/examples/sensor-control/api_how_to.h#L29) and then follow its code path?

Reply:
> I will not have time to review this properly before next week, but why do we need to set configs/parameters/defines at all for this? Can we not automatically detect which camera has been plugged in using [query_devices](https://github.com/IntelRealSense/librealsense/blob/master/examples/sensor-control/api_how_to.h#L29) and then follow its code path?

We certainly could do that but my reasoning was that the user may have multiple cameras attached to the system and would want to specify which model to use. Once #198 is added, it would be easier to refactor and allow the user to specify a serial number, or configure image size/FPS, accel/gyro rates, etc. for their use case. Right now, the goal would be to make it relatively simple to switch between the T26X and the currently supported RGB-D series streams. 
Open to suggestions  or other ideas on this.